scripts/files/fs_s3.py

```diff
@@ -0,0 +1,101 @@
+from typing import TYPE_CHECKING
+
+import boto3
+import botocore
+from linz_logger import get_log
+
+from scripts.aws.aws_helper import get_credentials, parse_path
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3.service_resource import S3ServiceResource
+else:
+    S3ServiceResource = object
+
+
+def get_auth_s3_session(bucket_name: str) -> S3ServiceResource:
+    """Return a boto3 S3 Resource with the AWS credentials for a bucket.
+
+    Args:
+        bucket_name (str): The name of the bucket.
+
+    Returns:
+        S3ServiceResource: The boto3 S3 Resource.
+    """
+    credentials = get_credentials(bucket_name)
+    session = boto3.Session(
+        aws_access_key_id=credentials.access_key,
+        aws_secret_access_key=credentials.secret_key,
+        aws_session_token=credentials.token,
+    )
+    s3: S3ServiceResource = session.resource("s3")
+    return s3
+
+
+def write(destination: str, source: bytes, needs_credentials: bool = False) -> None:
+    """Write a source (bytes) in a AWS s3 destination (path in a bucket).
+
+    Args:
+        destination (str): The AWS S3 path to the file to write.
+        source (bytes): The source file in bytes.
+        needs_credentials (bool, optional): Tells if credentials are needed. Defaults to False.
+    """
+    if source is None:
+        get_log().error("write_s3_source_none", path=destination, error="The 'source' is None.")
+        raise Exception("The 'source' is None.")
+    bucket_name, key = parse_path(destination)
+    key = key[1:]
```

Comment:
this is a odd statement why is this needed?

Reply:
To remove the leading `/`. This will be put in the `parse_path` method instead.